### PR TITLE
fix: fix bug with trailing version info not being properly stripped

### PR DESCRIPTION
**Pull request by bcpeinhardt**
Fixes a bug where the version info was not being stripped properly from documentation links.

### CODE REVIEW

File: codersdk/deployment_test.go [hunk 1]
```diff
@@ -565,3 +565,38 @@ func TestPremiumSuperSet(t *testing.T) {
...
+		tc := tc
+
+		t.Run(tc.Name, func(t *testing.T) {
+			t.Parallel()
+
+			stripped := codersdk.RemoveTrailingVersionInfo(tc.Version)
+			if !assert.Equal(t, tc.ExpectedAfterStrippingInfo, stripped) {
+				// Log relevant information if the assertion fails
+				t.Logf("Test failed for case: %s\nExpected: %s\nGot: %s", tc.Name, tc.ExpectedAfterStrippingInfo, stripped)
+			}
```
Comment (ethanndickson) [inline codersdk/deployment_test.go:598]:
```suggestion
			require.Equal(t, tc.ExpectedAfterStrippingInfo, stripped)
```

File: codersdk/deployment_test.go [hunk 2]
```diff
@@ -565,3 +565,38 @@ func TestPremiumSuperSet(t *testing.T) {
...
+			ExpectedAfterStrippingInfo: "v2.16.0",
+		},
+		{
+			Version:                    "v2.16.0-devel+683a720)",
+			ExpectedAfterStrippingInfo: "v2.16.0",
+		},
+	}
+
+	for _, tc := range testCases {
+		// Is this still necessary?
```
Comment (ethanndickson) [inline codersdk/deployment_test.go:588]: Unfortunately yes :(
https://stackoverflow.com/questions/26692844/captured-closure-for-loop-variable-in-go

**Review by ethanndickson — APPROVED** [review]

File: codersdk/deployment_test.go [hunk 3]
```diff
@@ -565,3 +565,38 @@ func TestPremiumSuperSet(t *testing.T) {
 	require.NotContains(t, enterprise.Features(), "", "enterprise should not contain empty string")
 	require.NotContains(t, premium.Features(), "", "premium should not contain empty string")
 }
+
+func TestRemoveTrailingVersionInfo(t *testing.T) {
+	t.Parallel()
+
+	testCases := []struct {
+		Name                       string
```
Comment (ethanndickson) [inline codersdk/deployment_test.go:573]:
```suggestion
```

Comment (bcpeinhardt) [reply codersdk/deployment_test.go:573]: I forgot to name the tests 🤦 zero values smh.

File: site/src/theme/icons.json [hunk 4]
```diff
@@ -1,94 +0,0 @@
-[
```
Comment (ethanndickson) [inline site/src/theme/icons.json:1]: accidental inclusion?

**Review by ethanndickson — APPROVED** [review]

File: codersdk/deployment.go [hunk 5]
```diff
@@ -804,8 +804,12 @@ func DefaultSupportLinks(docsURL string) []LinkConfig {
 	}
 }
 
+func RemoveTrailingVersionInfo(v string) string {
```
Comment (aslilac) [inline codersdk/deployment.go:807]: I would suggest...

- making this function private (lowercase)
- making a new deployment_internal_test.go file that declares `package codersdk` (and not `codersdk_test`)
- moving the tests you added in there

**Review by aslilac — APPROVED** [review]

**Pull request merged**